src/tpm2/ImplementationPCClient.h

```diff
@@ -0,0 +1,1222 @@
+/*    FITNESS FOR A PARTICULAR PURPOSE, ACCURACY, COMPLETENESS, OR 		*/
+/*    NONINFRINGEMENT OF INTELLECTUAL PROPERTY RIGHTS, OR ANY WARRANTY 		*/
+/*    OTHERWISE ARISING OUT OF ANY PROPOSAL, SPECIFICATION OR SAMPLE.		*/
+/*										*/
+/*  - Without limitation, TCG and its members and licensors disclaim all 	*/
+/*    liability, including liability for infringement of any proprietary 	*/
+/*    rights, relating to use of information in this specification and to the	*/
+/*    implementation of this specification, and TCG disclaims all liability for	*/
+/*    cost of procurement of substitute goods or services, lost profits, loss 	*/
+/*    of use, loss of data or any incidental, consequential, direct, indirect, 	*/
+/*    or special damages, whether under contract, tort, warranty or otherwise, 	*/
+/*    arising in any way out of use or reliance upon this specification or any 	*/
+/*    information herein.							*/
+/*										*/
+/*  (c) Copyright IBM Corp. and others, 2016 - 2018				*/
+/*										*/
+/********************************************************************************/
+
+/* A.2 Implementation.h */
+#ifndef _IMPLEMENTATION_PCCLIENT_H_
+#define _IMPLEMENTATION_PCCLIENT_H_
+
+#include    "TpmBuildSwitches.h"
+#include    "BaseTypes.h"
+#include    "TPMB.h"
+#undef TRUE
+#undef FALSE
+#ifndef MAX
+#define MAX(a, b) ((a) > (b) ? (a) : (b))
+#endif
+#ifndef MIN
+#define MIN(a, b) ((a) < (b) ? (a) : (b))
+#endif
+/* From TPM 2.0 Part 2: Table 4 - Defines for Logic Values */
+#define  TRUE     1
+#define  FALSE    0
+#define  YES      1
+#define  NO       0
+#define  SET      1
+#define  CLEAR    0
+/* From Vendor-Specific: Table 1 - Defines for Processor Values */
+#if defined __FreeBSD__ || defined __DragonFly__   /* libtpms added begin */
+# include <sys/endian.h>
+#elif defined __APPLE__
+# include <libkern/OSByteOrder.h>
+#else
+# include <endian.h>
+#endif
+#if defined __linux__ || defined __CYGWIN__
+ #if __BYTE_ORDER == __LITTLE_ENDIAN
+  #define  BIG_ENDIAN_TPM       NO
+ #endif
+ #if __BYTE_ORDER == __BIG_ENDIAN
+  #define  BIG_ENDIAN_TPM       YES
+ #endif
+#elif defined __OpenBSD__ || defined __FreeBSD__ || defined __NetBSD__ \
+   || defined __DragonFly__
+ #if _BYTE_ORDER == _LITTLE_ENDIAN
+  #define  BIG_ENDIAN_TPM       NO
+ #endif
+ #if _BYTE_ORDER == _BIG_ENDIAN
+  #define  BIG_ENDIAN_TPM       YES
+ #endif
+#elif defined __APPLE__
+ #define  BIG_ENDIAN_TPM       NO
+#else
+ #error Unsupported OS
+#endif                                             /* libtpms added end */
+#ifndef  BIG_ENDIAN_TPM
+#define  BIG_ENDIAN_TPM       NO
+#endif
+#define  LITTLE_ENDIAN_TPM          !BIG_ENDIAN_TPM
+#define  MOST_SIGNIFICANT_BIT_0     NO
+#define  LEAST_SIGNIFICANT_BIT_0    !MOST_SIGNIFICANT_BIT_0
+#define  AUTO_ALIGN                 NO
+
+/* From TCG PC Client Platform 01.03v22: Table 2 PC Client TPM Algorithms */
+#define  RSA_KEY_SIZES_BITS         {2048}
```

Comment:
I heared about 3072 bits coming up. While this is not a problem of the pcclient config alone, I am wondering how to handle the bump to the next sizes.

Reply:
The PC Client config is supposed to be **minimal**, that means any optional features are disabled. Thus, if 3072 is not mandatory, it will be disabled in `ImplementationPCClient.h.`

Reply:
It will be bumped up to 3072 in the next version...

Reply:
Oh, you are right, I missed that. This PR is based on v1.03 anyway. Since the pcclient simulator will probably mainly be used for compatibility testing, less features is generally better (for now).

Reply:
we will have to base it on the latest accepted version...

Reply:
I think it is just as valid to support older versions. However, since I do not really have a strong opinion about this, I can update the values to v1.04 if you wish. I'll keep you posted.

Reply:
The new version is not far away and it doesn't make sense that the pcclient profile would be immediately obsoleted.